src/sitemapper.cr

```diff
@@ -19,36 +19,47 @@ module Sitemapper
     setting host : String, example: "https://mysite.com"
     setting sitemap_host : String? = nil
     setting max_urls : Int32 = 500
-    setting storage : Symbol = :local
+    setting storage : Sitemapper::Storage.class = Sitemapper::LocalStorage
     setting compress : Bool = true
-    setting aws_config : Hash(String, String)? = nil
+    setting aws_config : AwsStorageConfig? = nil
```

Comment:
Relying on the hash meant potential keys missing, or misspelling in keys and such. Moving to this struct at least ensures we have the proper settings